### DISCUSSION TITLE
Implementing an alert box showing during an experiment

### DISCUSSION THREAD

**Discussion by cedricbatailler** [jspsych/jsPsych#3094] [Q&A]
Hi there!

I had a random thought and realized I don't know how I would approach a problem. Let's say that you're conducting an experiment with repeated trials where a participant has to pick a correct response among two possible ones, as fast as they can. A risk, here, could be that the participant goes so fast that they basically doesn't care anymore about being correct.

To mitigate that risk, an approach could be to add a message shown in a trial if a specific condition is met (for example, four errors in a row). To do so, I think that the easiest way to implement it would be to add a conditional trial (jsPsych meaning) for each trial (experimental psychology meaning). 

But what if I wanted a [modal ](https://getbootstrap.com/docs/4.0/components/modal/) to appear instead? How would you approach this problem? 

I don't need a low-level answer to this question, but I am really interested in your insights. 

Cheers,
Cedric

**Comment by jodeleeuw** [comment]
Hey @cedricbatailler,

That's a great question. I agree that a conditional trial is probably the most "jsPsych-y" way to do this right now. Three other options come to mind:

1) If the modal were specific to a certain kind of trial then you could implement it as part of the plugin. 
2) The [extensions](https://www.jspsych.org/7.3/overview/extensions/) interface could probably support this as well. I think the tricky bit here would be triggering the modal at the appropriate time. If you can run the right logic in the trial's `on_load` handler then that might work.
3) You could probably just dump the code for the modal directly into the `on_load` handler. The advantage of the extension would be better abstraction and some options for state management.
